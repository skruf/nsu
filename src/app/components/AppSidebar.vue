<i18n>
{
  "en": {
    "events": "Events",
    "all": "All",
    "upcoming": "Upcoming",
    "history": "History",
    "categories": "Categories",
    "clubs": "Clubs",
    "classes": "Classes",
    "ranges": "Ranges",
    "records": "Records",
    "settings": "Settings"
  },
  "no": {
    "events": "Stevner",
    "all": "Alle",
    "upcoming": "Kommende",
    "history": "Avsluttet",
    "categories": "Kategorier",
    "clubs": "Klubber",
    "classes": "Klasser",
    "ranges": "Skyttebaner",
    "records": "Rekorder",
    "settings": "Innstillinger"
  }
}
</i18n>

<style lang="stylus">
#app
  padding-left 180px
  &.sidebar-open
    padding-left 64px
    i.menu-icon
      margin-right 0
    .app_aside
      width 64px !important
    .logo
      max-width 40px
    .el-menu
      padding-left 0
      text-align center
    .el-menu-item,
    .el-submenu__title
      margin-bottom 10px
    .menu-img
      width 100%
      margin-right 0

.el-submenu .el-menu-item
  padding-left 56px !important
  font-size 13px

.menu-img
  width 23px
  margin-right 8px
  filter invert(.5) sepia(1) saturate(200) hue-rotate(380deg)

.logo
  max-width 84px
  padding 14px 0
  margin-top 25px
  margin-bottom 10px

.el-button.collapse-button
  position fixed
  bottom 0
  left 0
  right 0
  border none
  background-color transparent
  box-shadow none
  &, &:active
    top auto
  i.menu-icon
    margin-right 0

.el-menu.el-menu--popup
  background-color var(--bg-color)

.app_aside
  width 180px !important
  display flex
  flex-direction column
  position absolute
  left 0
  top 0
  bottom 0
  background-color var(--bg-alt-color)
  box-shadow var(--box-shadow)

</style>

<template>
  <el-aside
    width="auto"
    class="app_aside"
  >
    <img
      class="logo mx-auto block"
      src="../../../static/img/nsu-logo.png"
    >

    <el-menu
      default-active="1-1"
      :collapse="isOpen"
    >
      <el-submenu index="1">
        <template slot="title">
          <img
            class="menu-img"
            src="../../../static/img/event-icon.svg"
          >
          <span>{{ $t("events") }}</span>
        </template>
        <el-menu-item
          index="1-1"
          @click="navTo('/events')"
        >
          {{ $t("all") }}
        </el-menu-item>
        <el-menu-item
          index="1-2"
          @click="navTo('/events?filter=upcoming')"
        >
          {{ $t("upcoming") }}
        </el-menu-item>
        <el-menu-item
          index="1-3"
          @click="navTo('/events?filter=history')"
        >
          {{ $t("history") }}
        </el-menu-item>
        <el-menu-item
          index="1-4"
          @click="navTo('/events/categories')"
        >
          {{ $t("categories") }}
        </el-menu-item>
      </el-submenu>

      <el-menu-item
        index="2"
        @click="navTo('/clubs')"
      >
        <img
          class="menu-img"
          src="../../../static/img/club-icon.svg"
        >
        <span>{{ $t("clubs") }}</span>
      </el-menu-item>

      <el-menu-item
        index="3"
        @click="navTo('/classes')"
      >
        <img
          class="menu-img"
          src="../../../static/img/class-icon.svg"
        >
        <span>{{ $t("classes") }}</span>
      </el-menu-item>

      <el-menu-item
        index="4"
        @click="navTo('/ranges')"
      >
        <img
          class="menu-img"
          src="../../../static/img/range-icon.svg"
        >
        <span>{{ $t("ranges") }}</span>
      </el-menu-item>

      <el-menu-item
        index="5"
        disabled
        @click="navTo('/records')"
      >
        <img
          class="menu-img"
          src="../../../static/img/record-icon.svg"
        >
        <span>{{ $t("records") }}</span>
      </el-menu-item>

      <el-menu-item
        index="6"
        @click="navTo('/settings')"
      >
        <img
          class="menu-img"
          src="../../../static/img/settings-icon.svg"
        >
        <span>{{ $t("settings") }}</span>
      </el-menu-item>
    </el-menu>

    <el-button
      class="button--text collapse-button"
      @click="toggle"
    >
      <i
        v-if="isOpen"
        class="menu-icon el-icon-arrow-right"
      />
      <i
        v-else
        class="menu-icon el-icon-arrow-left"
      />
    </el-button>
  </el-aside>
</template>

<script>
export default {
  name: "AppSidebar",

  props: {
    open: { type: Boolean, default: true }
  },

  data: function() {
    return {
      isOpen: this.open
    }
  },

  watch: {
    open(open) {
      this.isOpen = open
      this.$emit("update:open", open)
    }
  },

  methods: {
    toggle() {
      this.isOpen = !this.isOpen
      this.$emit("update:open", this.isOpen)
    },

    navTo(url) {
      this.$router.push(url)
    }
  }
}
</script>
