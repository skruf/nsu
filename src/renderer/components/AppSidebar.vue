<style lang="stylus">
#app
  padding-left 180px
  &.sidebar-open
    padding-left 64px
    i.menu-icon
      margin-right 0
    .app_aside
      width 64px !important
    .logo
      max-width 40px
    .el-menu
      padding-left 0
      text-align center

.logo
  max-width 84px
  padding 14px 0
  margin-top 25px
  margin-bottom 10px

.el-button.collapse-button
  position fixed
  bottom 0
  left 0
  right 0
  border none
  background-color transparent
  box-shadow none
  &, &:active
    top auto
  i.menu-icon
    margin-right 0

.el-menu.el-menu--popup
  background-color var(--bg-color)

.app_aside
  width 180px !important
  display flex
  flex-direction column
  position absolute
  left 0
  top 0
  bottom 0
  background-color var(--bg-alt-color)

</style>

<template>
  <el-aside
    width="auto"
    class="app_aside"
  >
    <img
      class="logo mx-auto block"
      src="https://norsksvartkruttunion.no/test/wp-content/uploads/2014/03/nsu-logo-full.png"
    >

    <el-menu
      default-active="1-1"
      :collapse="isOpen"
    >
      <el-submenu index="1">
        <template slot="title">
          <i class="menu-icon ion-logo-ionic" />
          <span>Events</span>
        </template>
        <el-menu-item
          index="1-1"
          @click="navTo('/events')"
        >
          All
        </el-menu-item>
        <el-menu-item
          index="1-2"
          @click="navTo('/events?filter=upcoming')"
        >
          Upcoming
        </el-menu-item>
        <el-menu-item
          index="1-3"
          @click="navTo('/events?filter=history')"
        >
          History
        </el-menu-item>
        <el-menu-item
          index="1-4"
          disabled
          @click="navTo('/events/categories')"
        >
          Categories
        </el-menu-item>
      </el-submenu>

      <el-menu-item
        index="2"
        @click="navTo('/classes')"
      >
        <i class="menu-icon ion-ios-nuclear" />
        <span>Classes</span>
      </el-menu-item>

      <el-menu-item
        index="3"
        @click="navTo('/clubs')"
      >
        <i class="menu-icon ion-ios-people" />
        <span>Clubs</span>
      </el-menu-item>

      <el-menu-item
        index="4"
        @click="navTo('/ranges')"
      >
        <i class="menu-icon ion-ios-people" />
        <span>Ranges</span>
      </el-menu-item>

      <el-menu-item
        index="5"
        disabled
        @click="navTo('/records')"
      >
        <i class="menu-icon ion-ios-star" />
        <span>Records</span>
      </el-menu-item>

      <el-menu-item
        index="6"
        @click="navTo('/settings')"
      >
        <i class="menu-icon ion-ios-settings" />
        <span>Settings</span>
      </el-menu-item>
    </el-menu>

    <el-button
      class="button--text collapse-button"
      @click="toggle"
    >
      <i
        v-if="isOpen"
        class="menu-icon ion-ios-expand"
      />
      <i
        v-else
        class="menu-icon ion-ios-contract"
      />
    </el-button>
  </el-aside>
</template>

<script>
export default {
  name: "AppSidebar",

  props: {
    open: { type: Boolean, default: true }
  },

  data: function() {
    return {
      isOpen: this.open
    }
  },

  watch: {
    open(open) {
      this.isOpen = open
      this.$emit("update:open", open)
    }
  },

  methods: {
    toggle() {
      this.isOpen = !this.isOpen
      this.$emit("update:open", this.isOpen)
    },

    navTo(url) {
      this.$router.push(url)
    }
  }
}
</script>
