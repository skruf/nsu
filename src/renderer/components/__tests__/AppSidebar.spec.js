import createLocalVue from "@/utils/tests/createLocalVue"
import mount from "@/utils/tests/mount"
import { Aside, Menu, Submenu, MenuItem, Button } from "element-ui"
import AppSidebar from "@/components/AppSidebar"

const localVue = createLocalVue([ Aside, Menu, Submenu, MenuItem, Button ])

describe("AppSidebar", () => {
  const build = (overrides = {}, deep = false) => {
    return mount(AppSidebar, localVue, overrides, deep)
  }

  // beforeEach(() => {
  //   jest.resetAllMocks()
  // })

  it("renders the component", () => {
    const wrapper = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have a form and an input", () => {
    const wrapper = build()
    expect(wrapper.find(Aside).exists()).toBe(true)
    expect(wrapper.find(Menu).exists()).toBe(true)
    expect(wrapper.find(Submenu).exists()).toBe(true)
    expect(wrapper.find(MenuItem).exists()).toBe(true)
    expect(wrapper.find(Button).exists()).toBe(true)
  })
})
