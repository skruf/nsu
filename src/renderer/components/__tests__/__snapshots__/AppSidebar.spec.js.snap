// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSidebar renders the component 1`] = `
<el-aside-stub width="auto" class="app_aside"><img src="https://norsksvartkruttunion.no/test/wp-content/uploads/2014/03/nsu-logo-full.png" class="logo mx-auto block">
  <el-menu-stub mode="vertical" defaultactive="1-1" menutrigger="hover" collapse="true" collapsetransition="true">
    <el-submenu-stub index="1" showtimeout="300" hidetimeout="300"><template></template>
      <el-menu-item-stub index="1-1">
        All
      </el-menu-item-stub>
      <el-menu-item-stub index="1-2">
        Upcoming
      </el-menu-item-stub>
      <el-menu-item-stub index="1-3">
        History
      </el-menu-item-stub>
      <el-menu-item-stub index="1-4" disabled="true">
        Categories
      </el-menu-item-stub>
    </el-submenu-stub>
    <el-menu-item-stub index="2"><i class="menu-icon ion-ios-nuclear"></i> <span>Classes</span></el-menu-item-stub>
    <el-menu-item-stub index="3"><i class="menu-icon ion-ios-people"></i> <span>Clubs</span></el-menu-item-stub>
    <el-menu-item-stub index="4"><i class="menu-icon ion-ios-people"></i> <span>Ranges</span></el-menu-item-stub>
    <el-menu-item-stub index="5" disabled="true"><i class="menu-icon ion-ios-star"></i> <span>Records</span></el-menu-item-stub>
    <el-menu-item-stub index="6" disabled="true"><i class="menu-icon ion-ios-settings"></i> <span>Settings</span></el-menu-item-stub>
  </el-menu-stub>
  <el-button-stub type="default" icon="" nativetype="button" class="button--text collapse-button"><i class="menu-icon ion-ios-expand"></i></el-button-stub>
</el-aside-stub>
`;
